app/src/main/java/de/tum/in/tumcampusapp/component/tumui/calendar/CalendarFragment.kt

```diff
@@ -514,6 +566,59 @@ class CalendarFragment :
             .show()
     }
 
+    private fun refreshMonthView() {
+        monthYearText.text = formatLocalDate(selectedDate)
+        val daysInMonth = daysInMonth(selectedDate)
+
+        val eventMap = calendarController.getEventsForMonth(selectedDate)
+
+        if (!::monthViewAdapter.isInitialized) {
+            monthViewAdapter = MonthViewAdapter(daysInMonth, eventMap)
+            monthRecyclerView.adapter = monthViewAdapter
+        } else {
+            monthViewAdapter.updateData(daysInMonth, eventMap)
+        }
+
+        val layoutManager: RecyclerView.LayoutManager = GridLayoutManager(requireContext(), 7)
+        monthRecyclerView.layoutManager = layoutManager
+    }
+
+    private fun daysInMonth(date: LocalDate): ArrayList<String> {
+        val daysInMonthArray: ArrayList<String> = ArrayList()
+        val yearMonth = YearMonth.of(date.year, date.monthOfYear)
+        val daysInMonth = yearMonth.lengthOfMonth()
+        val firstOfMonth = date.withDayOfMonth(1)
+        var dayOfWeek = firstOfMonth.dayOfWeek().get()
```

Comment:
I noticed that the order of the days is wrong. I think that's because `dayOfWeek() == 0 ` is not monday but sunday. 
```suggestion
        var dayOfWeek = firstOfMonth.dayOfWeek().get() - 1
```
That should solve the problem.

Reply:
```suggestion
        var dayOfWeek = firstOfMonth.dayOfWeek().get() - 1 // Monday is the first day of the week in Europe
```